server/src/main/java/io/crate/expression/scalar/ArrayCatFunction.java

```diff
@@ -38,12 +39,28 @@
 import io.crate.types.ArrayType;
 import io.crate.types.DataType;
 import io.crate.types.TypeSignature;
+import io.crate.types.UndefinedType;
 
 public class ArrayCatFunction extends Scalar<List<Object>, List<Object>> {
 
     public static final String NAME = "array_cat";
 
     public static void register(Functions.Builder module) {
+        // We need to register a concrete function implementation for the `NULL` arguments.
+        // Otherwise, the arguments type will cast to and `ARRAY(UNDEFINED)` which causes a validation error as
+        // the inner type of the array arguments cannot both be `UNDEFINED`.
+        module.add(
+            Signature.builder(NAME, FunctionType.SCALAR)
+                .argumentTypes(
+                    UndefinedType.INSTANCE.getTypeSignature(),
+                    UndefinedType.INSTANCE.getTypeSignature())
+                .returnType(UndefinedType.INSTANCE.getTypeSignature())
+                .features(Feature.DETERMINISTIC, Feature.NULLABLE)
```

Comment:
```suggestion
                .features(Feature.DETERMINISTIC)
```

This isn't a strict nullable given that `select array_cat(null, array[1, 2]);` still returns a value. `nullable` can only be used if it returns null as soon as any argument is null. (We should probably rename it from `NULLABLE` to `STRICTNULL`)

Reply:
Hm but `array_cat(null, array[1, 2]);` would not match this function. What do I miss?

Reply:
Ah right - I was looking at the `evaluate` implementation, not at the `BinaryScalar` here.